runbot_travis2docker/models/runbot_build.py

```diff
@@ -175,14 +173,26 @@ def checkout(self, cr, uid, ids, context=None):
             if not (repo_name.startswith('https://') or
                     repo_name.startswith('git@')):
                 repo_name = 'https://' + repo_name
-            sys.argv = [
-                'travisfile2dockerfile', repo_name,
-                branch_short_name, '--root-path=' + t2d_path,
-            ]
+            sha = build.name
+            git_data = get_git_data(
+                repo_name, os.path.join(t2d_path, 'repo'), sha)
+            git_data['revision'] = branch_short_name
+            yml_content = git_data['content']
+            t2d_e = None
             try:
-                path_scripts = t2d()
-            except BaseException:  # TODO: Add custom exception to t2d
+                t2d_obj = Travis2Docker(
+                    yml_buffer=yml_content,
+                    work_path=os.path.join(t2d_path, 'script',
+                                           str(build.id) + "_" + sha[:7]),
+                    os_kwargs=git_data,
+                    copy_paths=[("~/.ssh", "$HOME/.ssh")],
+                )
+                path_scripts = t2d_obj.compute_dockerfile(
+                    skip_after_success=True)
+            except BaseException as t2d_e:
```

Comment:
Can you add the to-do into the ReadMe please? I don't like this line - BaseException should never be in an except - so it would be a shame to lose the to-do